x-pack/plugins/siem/public/plugin.tsx

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+
+import {
+  AppMountParameters,
+  CoreSetup,
+  CoreStart,
+  PluginInitializerContext,
+  Plugin as IPlugin,
+} from '../../../../src/core/public';
+import {
+  HomePublicPluginSetup,
+  FeatureCatalogueCategory,
+} from '../../../../src/plugins/home/public';
+import { DataPublicPluginStart } from '../../../../src/plugins/data/public';
+import { EmbeddableStart } from '../../../../src/plugins/embeddable/public';
+import { Start as NewsfeedStart } from '../../../../src/plugins/newsfeed/public';
+import { Start as InspectorStart } from '../../../../src/plugins/inspector/public';
+import { UiActionsStart } from '../../../../src/plugins/ui_actions/public';
+import { UsageCollectionSetup } from '../../../../src/plugins/usage_collection/public';
+import {
+  TriggersAndActionsUIPublicPluginSetup as TriggersActionsSetup,
+  TriggersAndActionsUIPublicPluginStart as TriggersActionsStart,
+} from '../../triggers_actions_ui/public';
+import { SecurityPluginSetup } from '../../security/public';
+import { APP_ID, APP_NAME, APP_PATH, APP_ICON } from '../common/constants';
+import { initTelemetry } from './lib/telemetry';
+import { KibanaServices } from './lib/kibana';
```

Comment:
Because we don't do tree-shaking this will inadvertently pull in the `kibana_react` plugin. It's currently an external module so this won't increase the bundle-size, but it would be better to have separate modules for KibanaServices used in the `plugin.ts` and the `kibana_react` and other dependencies used in `app.tsx`

Reply:
@rudolf I addressed this one specifically in dd1bcd6, but the `initTelemetry` and `serviceNowActionType` imports correspond to ~500kB and ~70kB worth of code, respectively. Without those, the plugin bundle is about 10kB (!). I created https://github.com/elastic/kibana/issues/64589 as a followup to this PR.